Task 10: Login

yevgenii-v opened the pull request:
- Authentication was created
- A new user can register a new account, log in, view information about himself
and log out of his account
- Logged user can not create new account or be logged again
- Guest user can not use profile or logout routes

yevgenii-v committed:
Task 10: Login
- Authentication was created
- A new user can register a new account,
log in, view information about himself
and log out of his account
- Logged user can not create new account
or be logged again
- Guest user can not use profile or logout
routes

SanycoWS commented on app/Http/Requests/Authentication/LoginRequest.php:
```diff
@@ -0,0 +1,22 @@
+    {
+        return [
+            'email'     => ['required', 'email', 'exists:users,email', 'max:255'],
+            'password'  => ['required', 'string', 'max:255'],
```
для пароля рекомендую додати більш складніші умови
Password::min(8)
                        ->mixedCase() # Потрібна хоча б одна велика і одна мала літери
                        ->numbers() # Потрібна хоча б одна цифра
                        ->symbols() # Потрібний хоча б один символ
це як приклад...

SanycoWS commented on app/Http/Requests/Authentication/LoginRequest.php:
```diff
@@ -0,0 +1,22 @@
+    public function rules(): array
+    {
+        return [
+            'email'     => ['required', 'email', 'exists:users,email', 'max:255'],
```
для емейлу рекомендую додавати додаткові умови. якщо користуватися стандартними які є в валідаторі https://laravel.com/docs/10.x/validation#rule-email то це email:rfc,dns

SanycoWS commented on app/Repositories/Authentication/Iterators/LoginUserIterator.php:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * @return PersonalAccessTokenResult
+     */
+    public function getBearerToken(): PersonalAccessTokenResult
```
цей метод не використовується і до даного класу його не потрібно додавати. даний клас має відображати тільки структуру вибірки із БД, а в тебе цього значення у вибірці немає)

yevgenii-v committed:
Task 10: Refactoring
- Add additional rule for password
in LoginRequest
- Remove unused argument in
LoginUserIterator constructor
